### ISSUE TITLE
Counting per classes and printing it on the final image?

### ISSUE THREAD

**Issue by juanbex** [ultralytics/yolov5#7136]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello 🥇 

I trained my model with custom dataset. I wrote some lines of code to get the numbers of object detected on a .jpg image.

`global count`
 `cv2.putText(im0,"Flowers: " + str(count), (20, 20), cv2.FONT_HERSHEY_SIMPLEX, 1, (25, 0, 0), 2, cv2.LINE_AA)`
 `cv2.imwrite(save_path, im0)`

Funtion:
`def count_obj():` 
    `global count `   
    `count += 1`

And I implemented it on 

`if save_img or save_crop or view_img:`
 ` c = int(cls)  # integer class`
 ` label = None if hide_labels else (names[c] if hide_conf else f'{names[c]} {conf:.2f}')`
` count_obj()`

Suppose I trained my model to detect 5 kinds of flowers. And I want to count them by classes to add them as text to my final image.

Someone could help me pls?

Here what I got without the counter by classes. 

![flower](https://user-images.githubusercontent.com/78883179/160025153-850525aa-0ad1-43f1-aabd-67601d5890b8.jpg)

**Comment by glenn-jocher** [comment]
@juanbex 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://docs.ultralytics.com/yolov5/tutorials/model_export) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # or yolov5m, yolov5l, yolov5x, etc.
# model = torch.hub.load('ultralytics/yolov5', 'custom', 'path/to/best.pt')  # custom trained model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.

results.xyxy[0]  # im predictions (tensor)
results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading) for details.

Good luck 🍀 and let us know if you have any other questions!
